include/openmc/mesh.h

```diff
@@ -427,7 +427,7 @@ class PeriodicStructuredMesh : public StructuredMesh {
   PeriodicStructuredMesh() = default;
   PeriodicStructuredMesh(pugi::xml_node node) : StructuredMesh {node} {};
 
-  void local_coords(Position& r) const override { r -= origin_; };
+  Position& local_coords(Position& r) const override { return r -= origin_; };
```

Comment:
I find this super confusing to have two overloads of `local_coords`, where the only difference is that one of them modifies its argument if it is not `const`. That side effect may seem surprising to a caller, who may assume that `local_coords` simply returns the position in the local coordinates. How about just having a single method with the following interface:
```C++
Position local_coords(const Position& r) const { return r - origin_; }
```
and if the caller wants to modify the value, they can do so explicitly.

Reply:
Yeah, you're right. We shouldn't have both. My only thought in favor of the in-place modification and returned reference was avoiding extra allocations, but doing some cursory tests with the `filter_mesh` regression tests it doesn't seem we are taking much of a hit there.